### DISCUSSION TITLE
Provide loading state when v7_startTransition is enabled and rendering is suspended

### DISCUSSION THREAD

**Discussion by dinnymate** [remix-run/react-router#10939] [Proposals]
When rendering is suspended for a long time (e.g. data fetching, though I know that loaders are the recommended way to do it, but sometimes...) the user doesn't see any indication of a navigation happening.

I know that in theory wrapping the function returned by `useNavigate` in `useTransition` would be a solution in some cases, but not when  navigation is initiated by the browser.

I'd like to suggest a feature where either `useNavigation().state` returns loading while the transition is pending, or the pending state is exposed separately.

I assume that React.startTransition should be replaced with useTransition for this to work. I tried making this modification myself but when switching to useTransition the transition doesn't work at all anymore.

**Comment by Diggsey** [comment]
I also ran into this issue - it would be great if we could show an indication when a transition is in progress.

**Comment by nythrox** [comment]
same here, this is pretty essential
